pb/api_test.go

```diff
@@ -0,0 +1,92 @@
+package pb
+
+import (
+	"math/rand"
+	"path"
+	"reflect"
+	"testing"
+	"testing/quick"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/bblfsh/sdk.v1/uast"
+)
+
+func TestMarshallingRoundTrip(t *testing.T) {
+	require := require.New(t)
+
+	f := func(ff *FileFixture) bool {
+		d, err := ff.File.Marshal()
+		if err != nil {
+			return false
+		}
+
+		of := &File{}
+		err = of.Unmarshal(d)
+		return err == nil
+	}
+
+	require.NoError(quick.Check(f, nil))
+}
+
+type FileFixture struct {
```

Comment:
This and `Generate` and `ValidGitModes` seems not to be used anywhere else - may be there is no need to export those?

Reply:
`Generate` implements Generator interface. I added comment.
And I made `ValidGitModes` private.